### PR TITLE
docs(nestjs): Update NestJS setup + event emitter

### PR DESCRIPTION
**Pull request by chargome**
**NOTE: wait for release `@sentry/nestjs` 8.39.0 before merging this**
- Updates nestjs setup to use `SentryGlobalGenericFilter` (closes https://github.com/getsentry/sentry-javascript/issues/14314)
- Defaults to ESM tab in getting started
- Documents https://github.com/getsentry/sentry-javascript/pull/14307 (event emitter)

### CODE REVIEW

**Review by andreiborza — APPROVED** [review]
@chargome I'd probably turn this into a draft PR so we don't accidentally merge it.

File: docs/platforms/javascript/guides/nestjs/features/event-emitter.mdx [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+description: "Learn about instrumenting NestJS event based services."
+---
+
+<Note>
+  The @nestjs/event-emitter module is auto-instrumented from version `@sentry/nestjs` 8.39.0 and up.
+</Note>
+
+The NestJS SDK wraps the `@OnEvent` decorator automatically to:
+
+- Create performance traces for event handler executions
```
Comment (coolguyzone) [inline docs/platforms/javascript/guides/nestjs/features/event-emitter.mdx:12]:
```suggestion
- Create performance traces for event handler executions.
```

File: docs/platforms/javascript/guides/nestjs/features/event-emitter.mdx [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+- Automatically capture any unhandled exceptions that occur in event handlers
```
Comment (coolguyzone) [inline docs/platforms/javascript/guides/nestjs/features/event-emitter.mdx:13]:
```suggestion
- Automatically capture any unhandled exceptions that occur in event handlers.
```

File: docs/platforms/javascript/guides/nestjs/features/event-emitter.mdx [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+- Maintain visibility into asynchronous event-driven flows
```
Comment (coolguyzone) [inline docs/platforms/javascript/guides/nestjs/features/event-emitter.mdx:14]:
```suggestion
- Maintain visibility into asynchronous event-driven flows.
```

**Review by coolguyzone — APPROVED** [review]
A couple suggestions, looks good to me! Thanks for adding.

**Pull request merged**